content/docs/overview.md

```diff
@@ -305,14 +305,55 @@ for key, value in some_map {
     fmt.println(key, value);
 }
 ```
-The iterated values are *copies* and cannot be written to. The following idiom is useful for iterating over a container in a by-reference manner:
+
+The iteration variables are _copies_ of the data in the array or map, and cannot be assigned to. See below for a way to get around this.
+
+**Note:** When iterating through a `string`, the characters will be `rune`s and not bytes. `for in` assumes the string to be encoded as UTF-8, and decodes each rune as it goes.
+
+### The Underscore (a.k.a unnamed variables)
+
+In certain situations you need to name a variable, but have no intent to use it.
+
+Consider:
 ```odin
-for _, i in some_slice {
-    some_slice[i] = something;
+for value, index in some_slice {
+    some_slice[index] = something;
 }
 ```
+`value` is never used here.
+
+In instances like this, you may use `_`:
+```odin
+for _, index in some_slice {
+    some_slice[index] = something;
+}
+```
+This is a useful idiom for iterating over a slice in a "by-reference" manner.
+
+You can also assign to it:
+```odin
+_ = f();
+_ = g();
+_ = h();
+```
+
+And you can use it with `:=` if you're also declaring at least one other variable.
+```odin
+_, ok := j(); // declares 'ok'
+_, ok = j();  // reuses 'ok'
+_ := k(); // nope
```

Comment:
Please don't write `nope`. Write that since there are no new variables being declared, this is not allowed within a procedure body. (This is allowed at file scope for different reasons)

Reply:
`nope` -> `no new variables declared`